### PR TITLE
[SPARK-50340][SQL] Unwrap UDT in INSERT input query

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

This is a new approach for https://github.com/apache/spark/pull/48251 .

We should unwrap UDT in the INSERT input query, so that we can treat them the same as the underlying SQL type, and allow writing them to the target table if the SQL type is compatible with table schema.

### Why are the changes needed?

allow writing UDT to the table that has the same SQL type.

### Does this PR introduce _any_ user-facing change?

Yes, writing UDT to table with same SQL type will be allowed.

### How was this patch tested?

new tests

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala [hunk 1]
```diff
@@ -530,16 +534,12 @@ object TableOutputResolver extends SQLConfHelper with Logging {
     } else {
       tableAttr.dataType
     }
-    lazy val outputField = if (isCompatible(tableAttr, queryExpr)) {
-      if (requiresNullChecks(queryExpr, tableAttr, conf)) {
-        val assert = AssertNotNull(queryExpr, colPath)
-        Some(Alias(assert, tableAttr.name)(explicitMetadata = Some(tableAttr.metadata)))
-      } else {
-        Some(queryExpr)
-      }
```
Comment (viirya) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala:539]: Is this change related? `AssertNotNull` is removed?

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala]: I misunderstood the code, I restored it now.

**Review by viirya — APPROVED** [review]

**Comment by cloud-fan** [comment]
I can't reproduce the group-by-all test failure locally, and it's definitely not related to this PR. Thanks for the review, merging to master!
